src/Data/FileEmbed/Extra.hs

```diff
@@ -22,3 +42,43 @@ embedFileIfExists inputPath = do
           reportWarning $ "File " <> inputPath <> " not found"
           pure (LitE $ StringL "")
     (_, Nothing) -> fail "No filepath provided"
+
+-- | Like `embedFile`, but prints out debug logging when
+-- @FOSSA_DEBUG_EMBED_FILE@ is set.
+embedFileL :: FilePath -> Q Exp
+embedFileL fp = qAddDependentFile fp >> runIO embedWithLogging >>= bsToExp
+  where
+    embedWithLogging = do
+      embedLogEnvVar <- lookupEnv "FOSSA_DEBUG_EMBED_FILE"
+      _ :: Either IOError () <- try $
+        when (isJust embedLogEnvVar) $ do
+          cwd <- getCurrentDirectory
+          putStrLn $ "CWD: " <> cwd
+
+          putStrLn $ "Embedding file: " <> fp
+          let isRel = isRelative fp
+          putStrLn $ "Relative?: " <> show isRel
+
+          let dirs = take (length (splitPath fp) - if isRel then 0 else 1) $ tail $ iterate takeDirectory fp
+          traverse_ ls dirs
+      B.readFile fp
+
+    ls :: FilePath -> IO ()
+    ls dir = do
+      files <- listDirectory dir
+      putStrLn $ "ls " <> show dir <> ": " <> show files
+
+-- See https://github.com/snoyberg/file-embed/blob/548430d2a79bb6f4cb4256768761071f59909aa5/Data/FileEmbed.hs#L184-L204
+bsToExp :: B.ByteString -> Q Exp
+bsToExp bs =
+  pure $
+    VarE 'unsafePerformIO
+      `AppE` ( VarE 'unsafePackAddressLen
+                `AppE` LitE (IntegerL $ fromIntegral $ B8.length bs)
+                `AppE` LitE
+                  ( bytesPrimL
+                      ( let B.PS ptr off sz = bs
+                         in mkBytes ptr (fromIntegral off) (fromIntegral sz)
+                      )
+                  )
+             )
```

Comment:
[blocking]: I'm going to look into doing this without unsafe code.  Unless there's an unavoidable reason to do unsafe, I'd rather take some extra time to try to remove it.